zebrad/src/commands/start/sync.rs

```diff
@@ -14,37 +14,32 @@ use zebra_consensus::checkpoint;
 use zebra_network::{self as zn, RetryLimit};
 use zebra_state::{self as zs};
 
-pub struct Syncer<ZN, ZS, ZV>
-where
-    ZN: Service<zn::Request>,
-{
-    pub peer_set: ZN,
-    pub state: ZS,
-    pub verifier: ZV,
-    pub retry_peer_set: Retry<RetryLimit, ZN>,
-    pub prospective_tips: HashSet<BlockHeaderHash>,
-    pub block_requests: FuturesUnordered<ZN::Future>,
-    pub fanout: NumReq,
-}
+// XXX in the future, we may not be able to access the checkpoint module.
+const FANOUT: usize = checkpoint::MAX_QUEUED_BLOCKS_PER_HEIGHT;
+/// Controls how far ahead of the chain tip the syncer tries to download before
+/// waiting for queued verifications to complete. Set to twice the maximum
+/// checkpoint distance.
+const LOOKAHEAD_LIMIT: usize = 2 * 2_000;
 
-impl<ZN, ZS, ZC> Syncer<ZN, ZS, ZC>
+#[derive(Debug)]
+pub struct Syncer<ZN, ZS, ZV>
 where
-    ZN: Service<zn::Request> + Clone,
+    ZN: Service<zn::Request, Response = zn::Response, Error = Error> + Send + Clone + 'static,
+    ZN::Future: Send,
+    ZS: Service<zs::Request, Response = zs::Response, Error = Error> + Send + Clone + 'static,
+    ZS::Future: Send,
+    ZV: Service<Arc<Block>, Response = BlockHeaderHash, Error = Error> + Send + Clone + 'static,
+    ZV::Future: Send,
```

Comment:
Trait bounds are added to the struct itself since we never want to create one without these bounds, and adding them to the struct rather than an impl gives clearer error messages (can't construct the struct rather than can't use some method)